libraries/Servo/src/Servo.cpp

```diff
@@ -74,29 +79,31 @@ uint8_t Servo::attach(int pin, uint16_t minUs, uint16_t maxUs)
   _maxUs = max((uint16_t)250, min((uint16_t)3000, maxUs));
   _minUs = max((uint16_t)200, min(_maxUs, minUs));
 
-  write(_valueUs);
+  write(value);
 
   return pin;
 }
 
 void Servo::detach()
 {
   if (_attached) {
+    startWaveform(_pin, 0, REFRESH_INTERVAL, 1);
+    delay(REFRESH_INTERVAL / 1000); // long enough to complete active period under all circumstances.
     stopWaveform(_pin);
     _attached = false;
-    digitalWrite(_pin, LOW);
+    _valueUs = DEFAULT_NEUTRAL_PULSE_WIDTH;
   }
 }
 
 void Servo::write(int value)
 {
-  // treat values less than 544 as angles in degrees (valid values in microseconds are handled as microseconds)
+  // treat values less than _minUs as angles in degrees (values equal or larger are handled as microseconds)
   if (value < _minUs) {
     // assumed to be 0-180 degrees servo
     value = constrain(value, 0, 180);
-    // writeMicroseconds will contrain the calculated value for us
-    // for any user defined min and max, but we must use default min max
     value = improved_map(value, 0, 180, _minUs, _maxUs);
+  } else if (value > _maxUs) {
+    value = _maxUs;
```

Comment:
Good bugfix!, however, this should really be done in `writeMicroseconds` because it's still possible to send in an illegal variable via that API  Move that check into the lower-lever routine (probably ensuring it's >=min and <=max) and you're good.

Reply:
@earlephilhower This API is a bit silly, now I have to check for _minUs twice, but if we are about protecting servos, let's do it right. I'm also setting the min and max DEFAULTs to 1000 and 2000 per the spec now, damnit :-)

Reply:
> ...have to check for minus 2x...

Well, the 1st one in this fcn. is just a bad API choice they made.  "I have an int, so why not silently use it as a degree or as a us value" seems silly vs adding a `writeDegrees` and making everything explicit.  In the lowest level `writeMicroseconds`, value===us always, though, so that's the only spot where "val<minUs" is an error.